app/queries/servers/scheduled_post.ts

```diff
@@ -68,3 +71,24 @@ export const observeScheduledPostCountForThread = (database: Database, rootId: s
         Q.where('root_id', rootId),
     ).observeCount();
 };
+
+export const observeScheduledPostEnabled = (database: Database) => {
+    const isScheduledPostConfigEnabled = observeConfigBooleanValue(database, 'ScheduledPosts');
+    const isLicensed = observeLicense(database);
+
+    const isScheduledPostEnabled = combineLatest([isScheduledPostConfigEnabled, isLicensed]).pipe(
+        switchMap(([isEnabled, license]) => {
+            if (license?.IsLicensed === 'true') {
+                return of$(isEnabled);
+            }
+            return of$(false);
+        }),
+        distinctUntilChanged(),
+    );
+
+    return isScheduledPostEnabled;
+};
+
+export const getIsScheduledPostEnabled = async (database: Database) => {
+    return (await getConfigValue(database, 'ScheduledPosts')) === 'true' && (await getLicense(database))?.IsLicensed === 'true';
+};
```

Comment:
Missing tests.

Reply:
@larkox we have to write the test for the complete file. It's already tracked here: https://mattermost.atlassian.net/browse/MM-62286